### DISCUSSION TITLE
Continuing on errors

### DISCUSSION THREAD

**Discussion by sophiajt** [pest-parser/pest#711] [Q&A]
Hi all,

Great work on Pest. I've been enjoying trying it out the last few weeks. We're actively talking about using it in Nushell as well.

Something I wasn't sure about, and couldn't find in the docs, is how to continue parsing after a parse error has occurred. This is pretty common in programming languages where the language needs to handle IDE support or other interactive cases (like a REPL). While the user is typing, it's likely the source is in a broken state, but you still want actions like completions to work correctly.

Is it possible in Pest to configure the parser to continue on errors and log the errors it finds?

Apologies if this is a duplicate question. I looked around but didn't see anything.

JT

**Comment by tomtau** [comment]
AFAIK there isn't an immediate easy configuration for this. (At least judging from that fact that, for example, IntelliJ plugin doesn't use pest's original meta-grammar, but reimplements it: https://github.com/pest-parser/intellij-pest/tree/master/grammar is that correct @ice1000?)
I think in the existing codebase, there may be hacky ways to achieve it depending on a grammar or by manually writing an equivalent of `impl Parser<Rule>` instead of generating it... But I assume the ideal solution here is that it could be generated: one issue is that `parse` is expected to return `Result<Pairs<R>, Error<R>>`, but here, it'd seem more appropriate to have `Vec<Error<R>>`...  @CAD97 @dragostis was this discussed before?

(One side-issue is incremental parsing, because one wouldn't want to fully re-parse on every small change in large files.)

**Reply by ice1000** [reply]
That is correct

**Comment by Gipson62** [comment]
I have the same issue, and it bothers me because having just one syntax error for multiple lines of code seems irrelevant. If there's a straightforward solution, I'd be glad to know it. I've been wanting to receive multiple errors for a while now. Currently, I'm creating a basic syntax in Pest and then adding additional checks in the compiler to generate more errors.
